port.js

```diff
@@ -464,9 +471,13 @@ Port.prototype.pipe = function pipe(stream, context) {
     var encode = this.encode(context);
     var decode = this.decode(context);
     var port = this;
+    var queueSize = function() {};
+    if (this && this.counter) {
+        queueSize = this.counter('gauge', `qsz-${this.config.id}`, 'Queue size');
```

Comment:
There is no need to put config.id in the counter name, as port already takes care of defining a measurement named after it. You can just put 'q' as name of the counter. Counter names should be short, as we do not want to generate big network packets.